### PR TITLE
[vidio] Add login support

### PR DESCRIPTION
**Pull request by MinePlayersPE**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Adds supports for logins via `--username` and `--password`
Note that while I have tested it with accounts, I'm not able to test it using account with a premier subscription as Google/Facebook logins are out of scope for this PR... and those are the only ones I have

### CODE REVIEW

**Comment by MinePlayersPE** [comment]
brb fixing netrc test

**Comment by MinePlayersPE** [comment]
done

File: yt_dlp/extractor/vidio.py [hunk 1]
```diff
@@ -46,10 +49,44 @@ class VidioIE(InfoExtractor):
...
+        login_page = self._download_webpage(
+            self._LOGIN_URL, None, 'Downloading log in page')
+
+        def is_logged():
+            return bool(self._download_json(
+                'https://www.vidio.com/interactions.json', None, False).get('current_user'))
+
+        # already logged in
+        if is_logged():
+            return
```
Comment (pukkandan) [inline yt_dlp/extractor/vidio.py:69]:
```suggestion
        def is_logged_in():
            resp = self._download_json(
                'https://www.vidio.com/interactions.json', None, 'Checking if logged in', fatal=False) or {}
            return bool(resp.get('current_user'))

        if is_logged_in():
            return
```

Comment (MinePlayersPE) [reply yt_dlp/extractor/vidio.py]: done, also pushed the check before the login page download

**Pull request merged**